sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -697,12 +735,17 @@ def renew_lock(self):
         # type: () -> None
         """Renew the message lock.
 
-        This will maintain the lock on the message to ensure
-        it is not returned to the queue to be reprocessed. In order to complete (or otherwise settle)
-        the message, the lock must be maintained. Messages received via ReceiveAndDelete mode are not
-        locked, and therefore cannot be renewed. This operation can also be performed as a threaded
-        background task by registering the message with an `azure.servicebus.AutoLockRenew` instance.
-        This operation is only available for non-sessionful messages.
+        This will maintain the lock on the message to ensure it is not returned to the queue 
+        to be reprocessed. 
+        
+        In order to complete (or otherwise settle) the message, the lock must be maintained, 
+        and cannot already have expired; an expired lock cannot be renewed.
+
+        Messages received via ReceiveAndDelete mode are not locked, and therefore cannot be renewed. 
+        This operation is only available for non-sessionful messages as well.
+
+        Lock renewal can be performed as a background task by registering the message with an 
```

Comment:
Just curious - still thinking of making this a built-in operation and not an external class?

Reply:
Yes; but it will necessarily come after we settle on an iterator->"getMessageIterator()/ReceiveForever" approach, since it would ideally be consistent for both receive modes.

there's also the consideration that it spins up concurrent threads somewhat-implicitly but it could be well declared in the docstring and thus slightly less something-to-be-avoided.